resource_obmcs_core_subnet_test.go

```diff
@@ -17,15 +18,26 @@ func TestAccResourceCoreSubnetCreate_basic(t *testing.T) {
 		data "oci_identity_availability_domains" "ADs" {
 			compartment_id = "${var.compartment_id}"
 		}
-		
+
 		resource "oci_core_virtual_network" "t" {
 			cidr_block     = "10.0.0.0/16"
 			compartment_id = "${var.compartment_id}"
 			display_name   = "network_name"
 		}`
 
+	commonSubnetParams := `
+					availability_domain = "${data.oci_identity_availability_domains.ADs.availability_domains.0.name}"
+					compartment_id = "${var.compartment_id}"
+					vcn_id = "${oci_core_virtual_network.t.id}"
+					security_list_ids = ["${oci_core_virtual_network.t.default_security_list_id}"]
+					route_table_id = "${oci_core_virtual_network.t.default_route_table_id}"
+					dhcp_options_id = "${oci_core_virtual_network.t.default_dhcp_options_id}"
+					cidr_block = "10.0.2.0/24"`
+
```

Comment:
Lets discuss our approach to templatizing reusable config blocks, stuff in provider_test.go is not maintainable and we can probably come up with something like this but more systematic, modular and granular across the board.